lib/R/Mbio/beta_div.R

```diff
@@ -0,0 +1,66 @@
+#   ## Inputs
+#   taxonomicLevel <- 'Class'
+#   k <- 20  # Keep pcoa dims
+#   method <- 'bray'
+#   verbose <- TRUE
+
+#   computeMessage <- ''
+  
+#   # Do argument checks
+#   # method = c('bray', 'jaccard', 'jsd')
+
+#   # Create otu table (samples x taxa)
+#   otu <- makeOTU(df, taxonomicLevel)
+#   plot.data::logWithTime(paste("Created OTU table with", NROW(otu), "samples and", (NCOL(otu)-1), "taxa."), verbose)
+
+#   # Compute beta div based on input
+#   if (identical(method, 'bray') | identical(method, 'jaccard')) {
+
+#     dist <- vegan::vegdist(otu[, -c('SampleID')], method=method, binary=TRUE)
+
+#   } else if (identical(method, 'jsd')) {
+
+#     # TO DO
```

Comment:
i saw the link. think itd probably not be too great an effort to provide our own jsd function which was a hybrid of the two solutions i saw there. both properly vectorized, and leveraging existing rcpp fxns.